### PR TITLE
create generic archonix scraper

### PR DESCRIPTION
**Pull request by terinpw**
- Move logic in ingest/us_pa_greene that is applicable to all archonix counties into ingest/archonix.
- Push case_number, bond_amount, and committed_by onto base person/record classes. 
- Add us_mt_gallatin using general Archonix classes

Follow up needed for archonix sites that differ slightly (ex extra options for seeing current inmates)

### CODE REVIEW

**Comment by arian487** [comment]
Overall this looks great, glad that things just work across vendors :)

I think this is mostly ready to go outside a few of the comments I made

**Comment by jessex** [comment]
I don't see your comments, @arian487! Is it possible you made them as a Github review but didn't submit the review yet? Or am I missing something?

File: recidiviz/ingest/archonix/archonix_person.py [hunk 1]
```diff
@@ -15,30 +15,24 @@
...
 """
 
 
 from google.appengine.ext import ndb
 from recidiviz.models.person import Person
 
 
-class UsPaGreenePerson(Person):
-    """A subclass of Person that adds Greene County, PA specific fields
+class ArchonixPerson(Person):
```
Comment (jessex) [inline recidiviz/ingest/archonix/archonix_person.py]: You know, I hadn't counted on having a _vendor-specific_ child class. I guess it makes perfect sense to do so given the current patterns, but the extra layer in the model hierarchy doesn't sit wonderfully with me, and makes me agree with @arian487 more about the desire to eliminate child models where possible. Of course, that's the promise of the universal schema :).

tl;dr: makes sense for now, just thinking aloud!

File: recidiviz/ingest/us_mt_gallatin/us_mt_gallatin_scraper.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+"""Scraper implementation for MT Gallatin County (Archonix)
+"""
+
+from recidiviz.ingest.archonix.archonix_scraper import ArchonixScraper
+
+class UsMtGallatinScraper(ArchonixScraper):
+    """Scraper for people in Gallatin County (MT) facilities."""
+    def __init__(self):
+        super(UsMtGallatinScraper, self).__init__('us_mt_gallatin')
```
Comment (jessex) [inline recidiviz/ingest/us_mt_gallatin/us_mt_gallatin_scraper.py]: Fantastic. Like Arian said, happy to see the complete consistency between these two Archonix counties.

File: recidiviz/ingest/archonix/archonix_scraper.py [hunk 3]
```diff
@@ -0,0 +1,541 @@
+# Recidiviz - a platform for tracking granular recidivism metrics in real time
```
[the remark this answers is not in the document]
Comment (jessex) [reply recidiviz/ingest/archonix/archonix_scraper.py]: Similarly, I'd like to move all region-specific code to `ingest/regions` but I'm happy to do that myself in a separate PR since it's a lot of movement and there are paths elsewhere in the codebase to be updated when that happens.

File: recidiviz/ingest/archonix/archonix_scraper.py [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# =============================================================================
+
+
+"""Scraper implementation for Archonix
+
+Region-specific notes:
```
Comment (jessex) [inline recidiviz/ingest/archonix/archonix_scraper.py]: `Vendor-specific notes`

File: recidiviz/ingest/archonix/archonix_scraper.py [hunk 5]
```diff
@@ -0,0 +1,541 @@
...
+            params: dict of parameters passed from the last scrape session.
+
+        Returns:
+            A list of Offense objects for the record.
+        """
+        offenses = []
+        charges = content.cssselect(
+            '[id=ctl00_ContentPlaceHolder1_gridCharges_ctl00]')[0]
+        body = charges.find('tbody')
+        # TODO(terinpw): We need to be smarter about this, as across counties
```
Comment (jessex) [inline recidiviz/ingest/archonix/archonix_scraper.py]: For TODOs like this that we don't intend to solve before this PR is merged, I think as a general guideline we should file an issue and then reference the issue number in the TODO, so future readers always have the context they need. So this would read, e.g. `# TODO (456): Handle county-specific columns more intelligently. Handle when generic scraper work is complete`

File: recidiviz/ingest/archonix/archonix_scraper.py [hunk 6]
```diff
@@ -0,0 +1,541 @@
...
+        for row in body:
+            offense = Offense()
+            offense.crime_class = row[0].text_content()
+            offense.description = row[1].text_content()
+            offense.case_number = row[2].text_content()
+            # Note that row[3] and row[4] represent the offense date and arrest
+            # date, But Archonix leaves this field blank.
+            bond_str = row[5].text_content()
+            # If there is no bond, it is filled with this character.
+            if bond_str.isspace():
```
[the remark this answers is not in the document]
Comment (terinpw) [reply recidiviz/ingest/archonix/archonix_scraper.py]: Rebased!

File: recidiviz/ingest/archonix/archonix_scraper.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (terinpw) [reply recidiviz/ingest/archonix/archonix_scraper.py]: I moved the vendor specific code into ingest/vendors, but I agree, let's leave the region clean up to a follow up.

File: recidiviz/ingest/archonix/archonix_scraper.py [hunk 7]
```diff
@@ -0,0 +1,541 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# =============================================================================
+
+
+"""Scraper implementation for Archonix
```
[the remark this answers is not in the document]
Comment (terinpw) [reply recidiviz/ingest/archonix/archonix_scraper.py]: Yup, done in the moved file.

File: recidiviz/ingest/archonix/archonix_scraper.py (hunk 4 above)
Comment (terinpw) [reply recidiviz/ingest/archonix/archonix_scraper.py]: Done.

**Comment by terinpw** [comment]
I've updated this for all previous comments. The failing checks seem unrelated (jailtracker file).

**Comment by jessex** [comment]
It does look like there are lint issues with the JailTracker scraper. Have you seen these, @zdg2102? Your own PR looked clean I thought. But there is one Archonix-related lint issue, too, @terinpw. The very first one about relative imports.

**Comment by terinpw** [comment]
Good catch - fixed the relative import issue.

**Comment by zdg2102** [comment]
Sorry about the lint issues, created https://github.com/Recidiviz/recidiviz-data/pull/173 which should fix them

**Comment by jessex** [comment]
Just merged that one from Zach. Once this is rebased it will kick off another Travis build and we should see it all clear.
